README.md

```diff
@@ -771,28 +777,85 @@ let numbers = create([Number], [1, 2, 3, 4]);
+
+## Structural Sharing
+
+A common performance optimization used by all reactive engines is to prevent re-renders for components who’s props have not changed. The most efficient way to determine if a value has not changed it to perform an exact equality check, for example: `prevValue === currentValue`. If the reference is the same, then consider the value unchanged. The Identity makes this possible with Microstates by internally managing how the Identity is constructed as a result of a transition. It will automatically determine which branches of microstates are unchanged and reuse previous identities for those branches.
+
+## Memoized Getters
+
+Microstates are immutable which makes it safe for us to memoize computations that are derived off their state. Identies will automatically memoize getters and return previously computed value when the microstate backing the identity has not changed. When a transition is invoked on the identity, the part of the identity tree that are changed will be re-created effectively invalidating the cache for the changed parts of the identity. The getters will be recomputed for state that is changed.
+
+## Debounce no-op transitions
+
+Identities automatically prevent unnecessary re-renders by debouncing transitions that do not change the value. This eliminates the need for `shouldComponentUpdate` hooks for pure components because it is safe to assume that a component that is re-rendering is re-rendering as a result of a transition that changed the value.
+
+## Identity Constructors
+
+Microstates comes with two Identity constructors: _Store_ and _Observable_. Observable will create a stream of identities. Next identity will be sent to the observer when an transition on an identity is called.
+
+### Store(microstate, callback)
+
+Store identity constructor takes two arguments: microstate and a callback. It returns an identity. When a transition is invoked on the identity, the callback will receive the next identity.
+
+```js
+import { Store, from, valueOf } from 'microstates';
+
+let initial = create(Number, 42);
+
+let last;
+
+last = Store(initial, next => (last = next));
+
+last.increment();
+//> undefined
+// callback will be invoked syncronously on transition
+
+// last here will reference the last
+last.increment();
+//> undefined
+
+valueOf(last);
+//> 44
+```
+
+The same mechanism can be used with React or any other reactive environment.
+
+```js
+import React from 'react';
+import { Store, create } from 'microstates';
+
+class Counter extends React.Component {
+  state = {
+    last: Store(create(Number, 42), next => this.setState({ last: next }))
+  };
+  render() {
+    let { last } = this.state;
+    return (
+      <button onClick={() => last.increment()}>Increment {last.state}</button>
+    );
+  }
+}
+```
+
+### Observable.from(microstate)
```

Comment:
I think that it might be useful to talk about when you might need to "upgrade" from using `Store` to a full-blown Observable library. This isn't really clear to me; from my perspective it feels like there are just two ways to do pretty much the same thing, but Observable uses an existing concept and is more powerful, as you also get all the benefits of the rest of whichever library you're using.

While `Store` has a very compelling use-case, like the React example above shows, it feels kind of weird to me that there are two different ways to subscribe to state changes. I'd almost rather there just be one, to cut down on API surface area and remove a layer of decisions that need to be made.

Reply:
Observable interface was designed specifically for Observable interop. It's not really meant to be used over Store, unless you're using Observables. Internally, Observable interop using `Identity` which is the same thing as a Store. 

I'll answer the rest of this the other message with questions.